install_zutils.py

```diff
@@ -0,0 +1,689 @@
+    try:
+        # Get list of submodules
+        submodules = subprocess.check_output(["git", "submodule", "status"], text=True).strip()
+
+        if not submodules:
+            return  # No submodules to check
+
+        # Check for uninitialized submodules (those starting with -)
+        uninitialized = [
+            line.strip() for line in submodules.split("\n") if line.strip().startswith("-")
+        ]
+
+        if uninitialized:
+            print_error(
+                "Git submodules are not initialized.\n"
+                "Please clone the repository with --recurse-submodules or run:\n"
+                "git submodule update --init --recursive\n\n"
+                "Alternatively, use --no_submodules to skip this check."
+            )
+
+    except subprocess.CalledProcessError:
+        print_error(
+            "Failed to check git submodules.\n"
+            "Please ensure you're in the zetta_utils repository root."
+        )
+
+
+def main():
+    console.print(Panel.fit("Zetta Utils Installer", style="bold cyan", subtitle="v1.0"))
+
+    parser = argparse.ArgumentParser(description="Install zetta_utils and dependencies")
+    parser.add_argument(
+        "--modules",
+        choices=["modules", "all"],
+        default="all",
+        help='Installation type: "modules" for basic installation or "all" for full installation',
+    )
+    parser.add_argument(
+        "--dockerfile",
+        action="store_true",
+        default=False,
+        help="Whether command is being evoked in a Dockerfile",
+    )
+    parser.add_argument(
+        "--skip_submodules", action="store_true", default=False, help="Skip submodules check"
+    )
+    parser.add_argument(
+        "--skip_apt", action="store_true", default=False, help="Skip apt modules install"
+    )
+    parser.add_argument(
+        "--skip_pip", action="store_true", default=False, help="Skip pip modules install"
+    )
+
+    args = parser.parse_args()
+
+    if not args.dockerfile:
+        check_ubuntu()
+        check_repository()
+        if not args.skip_submodules:
+            check_submodules()
+
+    if not args.skip_submodules:
+        apt_packages = [
+            "git",
+            "build-essential",
+            "wget",
+            "ffmpeg",
+            "libsm6",
+            "libxext6",
+            "curl",
+            "cmake",
+            "ninja-build",
+            "pkg-config",
+            "zstd",
+            "parallel",
+            "coreutils",
+            "libboost-system-dev",
+            "libboost-iostreams-dev",
+            "libjemalloc-dev",
+            "libtbb-dev",
+            "unixodbc-dev",
+        ]
+        run_command("sudo apt-get update", "Updating package lists")
+        apt_install_flags = '-y --no-install-recommends -o Dpkg::Options::="--force-confdef" -o Dpkg::Options::="--force-confold"'
+        run_command(
+            f"sudo apt-get install {apt_install_flags} {' '.join(apt_packages)}",
```

Comment:
`sudo` is not available in our current docker image (currently always logged in as `root`)